### PR TITLE
rapidyml: add  version 0.4.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **rapidyml/***

- update c4core/0.1.9 (which supports clang with libc++)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/rapidyaml/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,9 @@ conan_basic_setup(TARGETS)
 
 find_package(ryml REQUIRED CONFIG)
 
-set(CMAKE_CXX_STANDARD 11)
-
 add_executable(${PROJECT_NAME} test_package.cpp)
 target_link_libraries(${PROJECT_NAME} ryml::ryml)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)
```
Comment (SpaceIm) [inline recipes/rapidyaml/all/test_package/CMakeLists.txt:11]: you should update `cmake_minimum_required` to 3.8

File: recipes/rapidyaml/all/test_package/conanfile.py [hunk 2]
```diff
@@ -8,6 +8,9 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
+        if tools.Version(self.deps_cpp_info["rapidyaml"].version) >= "0.4.0":
+            cmake.definitions["RYML_USE_PARSE_IN_ARENA"] = True
+
```
Comment (SpaceIm) [inline recipes/rapidyaml/all/test_package/conanfile.py:13]:
```suggestion
```

File: recipes/rapidyaml/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -6,7 +6,9 @@ conan_basic_setup(TARGETS)
...
+if (RYML_USE_PARSE_IN_ARENA)
```
Comment (SpaceIm) [inline recipes/rapidyaml/all/test_package/CMakeLists.txt:12]:
```suggestion
if (ryml_VERSION VERSION_GREATER_EQUAL "0.4.0")
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]
simpler

File: recipes/rapidyaml/all/test_package/CMakeLists.txt (hunk 1 above)
Comment (toge) [reply recipes/rapidyaml/all/test_package/CMakeLists.txt:11]: Thank you.
I tend to forget cxx_std_11 added since 3.8.

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**